File: app/src/main/java/jp/cayhanecamel/notifysample/MainActivity.java
package jp.cayhanecamel.notifysample;

import android.content.Intent;
import android.content.res.Resources;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBarActivity;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.EditText;


public class MainActivity extends ActionBarActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(jp.cayhanecamel.notifysample.R.layout.activity_main);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.container, new PlaceholderFragment())
                    .commit();
        }
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    /**
     * A placeholder fragment containing a simple view.
     */
    public static class PlaceholderFragment extends Fragment {

        EditText ticker;
        EditText contentTitle;
        EditText contentText;
        EditText bigContentTitle;
        EditText summaryText;
        CheckBox bigPickture;
        CheckBox wearableBackgroundImage;

        public PlaceholderFragment() {
        }

        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup container,
                                 Bundle savedInstanceState) {
            View rootView = inflater.inflate(R.layout.fragment_main, container, false);

            rootView.findViewById(R.id.showNotification).setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    showNotification();
                }
            });

            ticker = (EditText) rootView.findViewById(R.id.ticker);
            contentTitle = (EditText) rootView.findViewById(R.id.contentTitle);
            contentText = (EditText) rootView.findViewById(R.id.contentText);
            bigContentTitle = (EditText) rootView.findViewById(R.id.bigContentTitle);
            summaryText = (EditText) rootView.findViewById(R.id.summaryText);
            bigPickture = (CheckBox) rootView.findViewById(R.id.bigPickture);
            wearableBackgroundImage = (CheckBox) rootView.findViewById(R.id.wearableBackgroundImage);


            return rootView;
        }

        private void showNotification() {
            Intent newIntent = new Intent(App.get(), MainActivity.class);
            newIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            newIntent.putExtra("isLaunchByNotification", true);

            NotificationDto dto = new NotificationDto();

            if (!TextUtils.isEmpty(ticker.getText())) {
                dto.ticker = ticker.getText().toString();
            }

            if (!TextUtils.isEmpty(contentTitle.getText())) {
                dto.contentTitle = contentTitle.getText().toString();
            }

            if (!TextUtils.isEmpty(contentText.getText())) {
                dto.contentText = contentText.getText().toString();
            }

            if (!TextUtils.isEmpty(bigContentTitle.getText())) {
                dto.bigContentTitle = bigContentTitle.getText().toString();
            }

            if (!TextUtils.isEmpty(summaryText.getText())) {
                dto.summaryText = summaryText.getText().toString();
            }


            dto.intent = newIntent;
            dto.id = 100;

            Resources r = App.get().getResources();
            if (bigPickture.isChecked()) {
                dto.bigPicture = BitmapFactory.decodeResource(r, R.drawable.big_picture);
            }

            if (wearableBackgroundImage.isChecked()) {
                dto.wearableBackgroundImage = BitmapFactory.decodeResource(r, R.drawable.wearable_background);
            }


            NotificationUtil.show(dto);
        }
    }
}

File: app/src/main/kotlin/jp/cayhanecamel/notifysample/MainActivity.kt
package jp.cayhanecamel.notifysample

import android.app.Notification
import android.app.PendingIntent
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.NotificationCompat
import android.support.v4.app.NotificationManagerCompat
import android.support.v7.app.ActionBarActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_main.*


class MainActivity : ActionBarActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(jp.cayhanecamel.notifysample.R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .add(R.id.container, PlaceholderFragment())
                    .commit()
        }
    }


    /**
     * A placeholder fragment containing a simple view.
     */
    class PlaceholderFragment : Fragment() {


        override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                                  savedInstanceState: Bundle?): View {
            val rootView = inflater!!.inflate(R.layout.fragment_main, container, false)

            rootView.findViewById(R.id.showNotification).setOnClickListener { showNotification() }

            return rootView
        }

        private fun showNotification() {

            // Intent の作成
            val contentIntent = PendingIntent.getActivity(
                    App.get(),
                    1000,
                    Intent(App.get(), NextActivity::class.java),
                    PendingIntent.FLAG_UPDATE_CURRENT)

            // LargeIcon の Bitmap を生成
            val largeIcon = BitmapFactory.decodeResource(App.get().resources, R.drawable.ic_launcher)

            // NotificationBuilderを作成
            val builder = NotificationCompat.Builder(App.get())
            builder.setContentIntent(contentIntent)

            builder.setSmallIcon(R.drawable.ic_launcher)
            builder.setLargeIcon(largeIcon)

            if (contentTitle.text.isNotEmpty()) {
                builder.setContentTitle(contentTitle.text)
            }

            if (contentText.text.isNotEmpty()) {
                builder.setContentText(contentText.text)
            }

            if (ticker.text.isNotEmpty()) {
                builder.setTicker(ticker.text)
            }

            if (bigPictureRadio.isChecked) {
                val r = App.get().resources
                val style = NotificationCompat.BigPictureStyle()
                style.bigPicture(BitmapFactory.decodeResource(r, R.drawable.big_picture))
                if (summaryText.text.isNotEmpty()) {
                    style.setSummaryText(summaryText.text)
                }
                builder.setStyle(style)

            } else if (bigTextRadio.isChecked) {
                val style = NotificationCompat.BigTextStyle()
                if (bigContentTitle.text.isNotEmpty()) {
                    style.setBigContentTitle(bigContentTitle.text)
                }
                if (summaryText.text.isNotEmpty()) {
                    style.setSummaryText(summaryText.text)
                }
                if(bigText.text.isNotEmpty()){
                    style.bigText(bigText.text)
                }
                builder.setStyle(style)

            } else if (inboxRadio.isChecked) {

            }


            // 通知するタイミング
            builder.setWhen(System.currentTimeMillis())
            // 通知時の音・バイブ・ライト
            builder.setDefaults(Notification.DEFAULT_SOUND or Notification.DEFAULT_VIBRATE or Notification.DEFAULT_LIGHTS)
            builder.setAutoCancel(true)


            // NotificationManagerを取得
            val notificationManager = NotificationManagerCompat.from(App.get())
            // Notificationを作成して通知
            notificationManager.notify(1000, builder.build())

        }

    }
}
